commit c1e76ed04ead5c7611cc71432d769580e204e946 (refs/changes/69/269/2)
Author: Marisa Emerson <mje@uq.edu.au>
Date:   2016-08-11T06:14:17+00:00 (3 years, 2 months ago)
    
    joyent/triton-cns#3 network wildcard entries doesn't work on multiple zones

diff --git a/lib/net-filter.js b/lib/net-filter.js
index 52a3a21..df51975 100644
--- a/lib/net-filter.js
+++ b/lib/net-filter.js
@@ -69,11 +69,12 @@ NetFilter.prototype._transform = function (vm, enc, cb) {
 				}
 			}
 		}
-		for (i = 0; i < zones.length; ++i) {
-			z = self.config[zones[i]];
-			if (z.networks.indexOf('*') !== -1 &&
-			    nic.zones.length === 0)
-				nic.zones.push(zones[i]);
+		if (nic.zones.length === 0) {
+			for (i = 0; i < zones.length; ++i) {
+				z = self.config[zones[i]];
+				if (z.networks.indexOf('*') !== -1)
+					nic.zones.push(zones[i]);
+			}
 		}
 	});
 
